Pmx qui marche pour Esial ;)

jbrv opened the pull request:
En gros :
- Petite passe sur les types  : @cho934, vérifie que j'ai pas oublié des casts dans les "print" sur l'I2C, sinon tu vas avoir des trucs bizarres... ^^
- Suppression du "Débug UDP" qui sert à rien et qui ne marche probablement pas, faut refaire en plus propre.
- Système de config au build, maintenant il faut créer un fichier `build/build_config.mk`. J'ai mis des configs pour Esial et PMX ; @cho934, vérifie/modifie celle de PMX. Ça entraine pas mal de modifs dans le code avec des `#if`/`#endif` un peu partout.
- Les pins des codeurs sont configurables directement dans le fichier de config. Pareil, j'ai modifié les fichiers de config PMX, mais c'est à vérifier. J'ai aussi viré les anciens fichiers de config, ils n'étaient pas à jour et ils ne servent à rien.

jbrv committed:
Petite correction/harmonisation des types

jbrv committed:
Système de configuration du build

Un fichier de config doit être créé: build_config/build_config.mk
Cf. build_config/build_config.default.mk pour la syntaxe

Permet d'activer/désactiver des bouts de code à la compilation.
Du coup, tous les bouts de code qui étaient désactivés avec des
commentaires /* ... */ le sont maintenant avec des #if/#endif
pour harmoniser le tout.

jbrv committed:
Ajout de la commande d'init via la série

jbrv committed:
Suppression du "débug UDP", pas maintenu, pas utilisé

jbrv committed:
Pins des codeurs dans le fichier de config

L'instanciation des codeurs est déplacée dans le Main.cpp, avec tout
le reste.
Les fichiers de configs sont mis à jour ; au passage, je vire les
fichiers de configs qui servent à rien, pas envie de les maintenir.

cho934 committed:
Modif pmrobotix ok

jbrv commented on asserv/main/Main.cpp:
```diff
@@ -459,7 +459,7 @@ void initAsserv(bool *prun)
         codeurs = new CodeursAVR(p5, p6, p7, p8);
 #   else
-#       error "Undefined encoder interface; check build configuration"
+//#       error "Undefined encoder interface; check build configuration"
```
Nope !
Si cette erreur là pète, c'est qu'il y a un problème de config, et que ça ne va probablement pas très bien marcher. S'il arrive à cette erreur, il n'a instancié aucun codeur, donc c'est pas bon.

je rappelle, il faut créer un fichier de config de build : `build/build_config.mk` avant de compiler. De toute façon, si tu as réussi à faire marcher l'asserv, cette erreur là ne doit pas arriver, sinon je ne comprends rien au préprocesseur C... :D

Trowa replied:
@jbrv perso je le mettrai dans config/ ton fichier de config pour définir les codeurs / moteurs / etc. au lieu de build/ qui théoriquement devrait être dans le .gitignore :)

jbrv replied:
Mm... ou un dossier build_config, parce que le dossier config est déjà utilisé pour les fichiers de config à l'exécution, et on risque de tout mélanger... ^^"

Trowa replied:
ok, ça me convient

jbrv replied:
Ah bah en fait c'est déjà le cas... xD
J'ai pas fait tant de bêtises que ça finalement... :D

**DONC**: il faut créer un fichier `build_config/build_config.mk`, voilà...!

jbrv replied:
Et du coup, je décommente les `#error`, ça doit marcher avec, sinon c'est qu'il y a un très gros problème

jbrv committed:
On remet les `#error`, ça doit marcher

Si les `#error`explosent, l'asserv ne marche pas, donc c'est une
jolie sécurité
